paper/paper.md

```diff
@@ -0,0 +1,119 @@
+
+Beyond this query functionality, the package also provides:
+
+- A fuzzy implementation validator that performs HTTP queries against remote or local OPTIMADE APIs, with test queries and expected responses generated dynamically based on the data served at the introspective `/info/` endpoint of the API implementation.
+- Entry "adapters" that can convert between OPTIMADE-compliant entries and the data models of popular Python libraries used widely in the materials science community: `pymatgen` [@pymatgen], ASE [@ASE], AiiDA [@AiiDA], and JARVIS [@JARVIS].
+
+# Use cases
+
+The package is currently used in production by three major data providers for materials science data:
+
+- The Materials Project [@MaterialsProject] uses `optimade-python-tools` alongside their existing API [@MAPI] and MongoDB database, providing access to highly-curated density-functional theory calculations across all known inorganic materials.
+`optimade-python-tools` handles filter parsing, database query generation and response validation by running the reference server implementation with minimal configuration.
+- NOMAD [@nomad] uses `optimade-python-tools` as a library to extend its existing web app with OPTIMADE API routes.
+ It uses the Elasticsearch implementation to filter millions of structures from published first-principles calculations provided by users and other projects.
+NOMAD also uses the filtering module in its own API to expose the OPTIMADE filter language in the user-centric web interface search bar.
+NOMAD uses a released version of `optimade-python-tools` and all necessary customization can be realized via configuration and sub-classing.
+- Materials Cloud [@MaterialsCloud] uses `optimade-python-tools` as a library to provide an OPTIMADE API entry to archived computational materials studies, created with the AiiDA [@AiiDA] Python framework and published through their archive.
+In this case, each individual study and archive entry has its own database and separate API entry.
+The Python classes within the `optimade` package have been extended to make use of AiiDA and its underlying [PostgreSQL](https://postgresql.org) storage engine.
+- The `optimade.adapters` module from the `optimade-python-tools` library is used in a graphical web client for OPTIMADE API implementations hosted on Materials Cloud [@MaterialsCloudClient].
+It allows users to search through OPTIMADE API implementations using widgets as well as typing in a filter string raw.
+The found structures can be inspected graphically and either converted and downloaded in formats provided by conversion functions in the `optimade.adapters` module or used in another Materials Cloud web tool, where the structure is automatically downloaded, converted, and transferred in the backgorund, partly using the `optimade.adapters` module.
```

Comment:
```suggestion
- The `optimade.adapters` module from the `optimade-python-tools` library is used in a graphical web client hosted on Materials Cloud [@MaterialsCloudClient].
It allows users to query OPTIMADE API implementations using user-friendly widgets as well as raw filter strings.
The client uses the registry of known OPTIMADE providers to allow easy switching between databases.
The crystal structures returned can be inspected visually and either downloaded in formats provided by conversion functions in the `optimade.adapters` module, or used seamlessly within other Materials Cloud web tools, where the structure is automatically validated and transferred in the background, partly using the `optimade.adapters` module.
```

Thoughts @CasperWA?

Reply:
Great re-write!

Should it be "raw OPTIMADE filter strings" maybe? Or is this implied well-enough?

Reply:
I think it is implied, I had "OPTIMADE" there originally but I think we say it enough times in the paper :sweat_smile:

Reply:
I will commit and rebuild the paper but feel free to change it again